core/publish-key/src/PublishKeyHandler.ts

```diff
@@ -0,0 +1,142 @@
+import { LambdaInterface } from "@aws-lambda-powertools/commons/types";
+import { Context } from "aws-lambda";
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Jwk, JWKSBody } from "../utils/Types";
+import { PutObjectCommand, S3Client } from "@aws-sdk/client-s3";
+import { NodeHttpHandler } from "@smithy/node-http-handler";
+import crypto from "crypto";
+import { KMS } from "@aws-sdk/client-kms";
+import { GetPublicKeyCommandOutput } from "@aws-sdk/client-kms/dist-types/commands/GetPublicKeyCommand";
+
+export const logger = new Logger({ serviceName: "PublishKeyHandler" });
+
+export class PublishKeyHandler implements LambdaInterface {
+    decryptionKeyID: string;
+    bucketName: string;
+    kmsClient: KMS;
+
+    constructor(decryptionKeyID: string | undefined, bucketName: string | undefined, kmsClient: KMS | undefined) {
+        if (!decryptionKeyID) {
+            throw new Error("Key ID is missing");
+        }
+        this.decryptionKeyID = decryptionKeyID;
+
+        if (!bucketName) {
+            throw new Error("bucketName is missing");
+        }
+        this.bucketName = bucketName;
+
+        if (!kmsClient) {
+            throw new Error("kmsClient is missing");
+        }
+        this.kmsClient = kmsClient;
+    }
+
+    readonly s3Client = new S3Client({
+        region: process.env.REGION,
+        maxAttempts: 5,
+        requestHandler: new NodeHttpHandler({
+            connectionTimeout: 5000,
+            requestTimeout: 5000,
+        }),
+    });
+
+    // PLEASE NOTE: THIS LAMBDA WILL OVERWRITE EVERYTHING IN THE JWKS.JSON FILE. It does not append the current decryption key details.
+    public async handler(event: Record<string, unknown>, context: Context): Promise<string | undefined> {
+        try {
+            logger.info(`Initiating lambda ${context.functionName} version ${context.functionVersion}`);
+            logger.debug(`Using key ${this.decryptionKeyID} and uploading to ${this.bucketName}`);
+
+            const jwksBody: JWKSBody = { keys: [] };
+
+            const decryptionKey: GetPublicKeyCommandOutput | null = await this.getFromKms();
+            const decryptionJwk: Jwk | null = this.convertToJwk(decryptionKey);
+            logger.info("Successfully obtained kmsKey as jwk");
+
+            if (decryptionJwk) {
+                jwksBody.keys.push(decryptionJwk);
+
+                const uploadParams = {
+                    Bucket: this.bucketName,
+                    Key: "jwks.json",
+                    Body: JSON.stringify(jwksBody),
+                    ContentType: "application/json",
+                };
+                logger.debug(`uploadParams = ${JSON.stringify(uploadParams)}`);
+
+                await this.s3Client.send(new PutObjectCommand(uploadParams));
+                logger.debug(`jwksBody = ${JSON.stringify(jwksBody)}`);
+
+                logger.info("Successfully Uploaded jwks.json to bucket");
```

Comment:
Has this replaced the object contents? Or uploaded a new version? Pretty sure the bucket is still versioned.

```suggestion
                logger.info("Successfully uploaded a new version of jwks.json to bucket");
```